### ISSUE TITLE
Derived fields in combination with line_format

### ISSUE THREAD

**Issue by ThommyH** [grafana/loki#9209]
Since posts #4021 #6614 got no reply, I am reopening the case.

```
Using Grafana+Loki+Tempo.
Lets say I have log lines like '... traceID=d2432...'. I can use derived field to create URL link to Tempo traces, great.
This works in Explore, but only until I'm not adding line_format that removes the "traceID=". It seems that derived fields are evaluated on the final result, in case of line_format on line_format result. Makes sense.
In other words, I cannot have nicely line_format-ted lines without traceID= and trace links at the same time?
```
Thank you!

**Comment by sharkymcdongles** [comment]
We are also seeing this. It would be nice if derived fields could be set before line_format calls.

**Comment by pavelvodrazka** [comment]
We're dealing with the same problem. It would be nice if the user could freely customize the line formatting without breaking the derived fields.

**Comment by danielrosa-zh** [comment]
@ThommyH , the same here, could you please open this on grafana gh ? I think it's more about grafana UI then on loki side

**Comment by deathsaber** [comment]
@ThommyH - I have the same issue and I was wondering if you were able to open this issue on grafana gh. If so, can you please share the issue # ?

**Comment by JakeCooper** [comment]
Massive+1 here

This kinda really messes with like, the ability to cleanly parse logs while being able to jump to traces

**Comment by Samreay** [comment]
+1, we've large lines of JSON logging and any attempt to clean things up using line_format means our telemetry trace_id tempo links vanish, and that's literally the whole point of us including the logs in the dashboard - so we can get to the relevant trace faster

**Comment by mladedav** [comment]
I'll just add that while this works, the derived field is available only if it is in the labels, i.e. if it was sent or if it was already processed with a function such as `| json` or `| logfmt`. If the logline is not parsed, it is not visible, because loki cannot find the field.

So I currently use both regex and this new feature so that there is always at least one of the derived fields available.

**Comment by artemik** [comment]
@ThommyH I don't think it should be closed given @mladedav note.

**Comment by omri-shilton** [comment]
We are also in need to be able to view the derived field while modifying the log line.
we are processing json logs so we must add a line filter to get the message.
